llvm/lib/Transforms/IPO/ThinLTOBitcodeWriter.cpp

```diff
@@ -487,9 +488,9 @@ void splitAndWriteThinLTOBitcode(
   // be used in the backends, and use that in the minimized bitcode
   // produced for the full link.
   ModuleHash ModHash = {{0}};
-  W.writeModule(M, /*ShouldPreserveUseListOrder=*/false, &Index,
+  W.writeModule(M, ShouldPreserveUseListOrder, &Index,
                 /*GenerateHash=*/true, &ModHash);
-  W.writeModule(*MergedM, /*ShouldPreserveUseListOrder=*/false, &MergedMIndex);
+  W.writeModule(*MergedM, ShouldPreserveUseListOrder, &MergedMIndex);
```

Comment:
There are some other calls to write the module in this function that presumably also need the change (and tests): see below where we write the module when requesting minimized bitcode (ThinLinkOS - for the opt tool this is set when `-thin-link-bitcode-file=` is specified).  Also at the beginning of this function there are some calls in the case where the ModuleId was empty - although it looks like that would only be the case if there are no externally visible symbols in the module, not sure if we ever would have or care about the uselistorder in that case?

Reply:
Unfortunately constants also have use lists that need ordering, so the symbols don't matter

Reply:
I don't know how to test all these paths. Most of these use are because the bitcode serialization is mixed in with transformations, and llvm-reduce really only wants pure bitcode serialization (e.g. why is AARGetter needed?). There should be a Transforms-free serialization path available, and llvm-reduce really shouldn't depend on the pass manager